config.go

```diff
@@ -0,0 +1,79 @@
+package carrot
+
+import (
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"sync"
+	"time"
+)
+
+var (
+	once   sync.Once
+	config Config
```

Comment:
Instead of a global variable let's have the config belong to the individual modules. We can do that by loading the config in `app.go`, then for each of the modules, we will pass the configuration argument for that module when it is initialized. For example,
```
func NewServer(clientPool *ClientPool, sessionStore SessionStore) *Server {
	/* ... */
}
```

would become 

``` 
func NewServer(clientPool *ClientPool, sessionStore SessionStore, config *config.Server) *Server {
	/* ... */
}
```

This will make the modules more portable (modules can run in isolation) and they can have their own validations for configuration when the time comes.